go/adapter/opentelemetry/adapter.go

```diff
@@ -0,0 +1,148 @@
+package opentelemetry
+
+import (
+	"context"
+	"log"
+	"time"
+
+	observe "github.com/dylibso/observe-sdk/go"
+	trace "go.opentelemetry.io/proto/otlp/trace/v1"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+)
+
+type OTLPProtocol string
+
+const (
+	GRPC OTLPProtocol = "grpc"
+	HTTP OTLPProtocol = "http/protobuf"
+)
+
+type OTelConfig struct {
+	ServiceName        string
+	EmitTracesInterval time.Duration
+	TraceBatchMax      uint32
+	Endpoint           string
+	Protocol           OTLPProtocol
+	ClientHeaders      map[string]string
+	AllowInsecure      bool
+
+	client otlptrace.Client
+}
+
+type OTelAdapter struct {
+	*observe.AdapterBase
+	Config *OTelConfig
+}
+
+func (a *OTelAdapter) UseCustomClient(client otlptrace.Client) {
+	if a.Config != nil {
+		a.Config.client = client
+	}
+}
```

Comment:
the `otlptrace.Client` interface makes for flexible customization of how to get data into a collector -- so this enables an end user to configure one using their own implementation. not used in the examples, but this prevents us from having to go config-happy in the `OTelConfig` struct.